### DISCUSSION TITLE
Using the same code to generate reports for different customers

### DISCUSSION THREAD

**Discussion by philipgladwin** [quarto-dev/quarto-cli#2527] [Q&A]
I have a dozen Customers that I would like to generate a report for.  The Customer's name is a variable in the code so that database tables can be queried for that Customer.  Working in Quarto which I'm enjoying very much.  Report is in the style of a book.  I'd like to run the report for one customer and keep it and run the second report for the next Customer.......Currently files are sent to a directory "_book" and are over written each time.  Is there any way to define in RStudio the name of the output directly during the render process?

Secondly how to I publish each report for each Customer to bookdown?

Thank you for any help.
Phil,

**Comment by jeremiahpslewis** [comment]
Check out environment variables here:

https://quarto.org/docs/projects/environment.html

https://quarto.org/docs/authoring/variables.html

**Reply by dragonstyle** [reply]
Also recommend checking out project profiles:

https://quarto.org/docs/projects/profiles.html
